geoutils/georaster/raster.py

```diff
@@ -1252,7 +1259,14 @@ def reproject(
 
         # If data is loaded, reproject the numpy array directly
         if self.is_loaded:
-            dst_data, dst_transformed = rio.warp.reproject(self.data, **reproj_kwargs)
+
+            # All masked values must be set to a nodata value for rasterio's reproject to work properly
+            # TODO: another option is to apply rio.warp.reproject to the mask to identify invalid pixels
+            if src_nodata is None and np.sum(self.data.mask) > 0:
+                raise ValueError("No nodata set, use `src_nodata`.")
+
+            # Mask not taken into account by rasterio, need to fill with src_nodata
+            dst_data, dst_transformed = rio.warp.reproject(self.data.filled(src_nodata), **reproj_kwargs)
```

Comment:
Tiens je connaissais pas `np.ma.filled`, y a tellement de choses différentes avec les masked_arrays ! C'est solide mais ça limite aussi l'accès aux utilisateurs, ça sera cool de finaliser rapidement les fonctions style `get_array` pour faciliter l'accès à une `np.ndarray` classique :)

Reply:
`np.ma.filled` is used quite often throughout our code 😉 
The problem is that there is not a unique solution to go from a masked array to a regular array. Nodata can be set to NaN, but only for compatible dtypes, or it can be set to nodata, but then it is not ignored in calculations.
Here it's simply because rasterio expects a regular array with gaps filled to the set nodata. No need for the user to know that.